### PR TITLE
Get stardoc working and update docs. #140

### PR DESCRIPTION
**Pull request by mfarrugi**
Attribute docstrings are now more clearly the canonical documentation, removed redundant copies elsewhere. README now points to generated docs.

Also added @docs//:rust-docs to .bazelci so that it is less likely to fall into disrepair again..

Had various issues with stardoc, but if we comment out loads in `rustc.bzl` it does work and produced `index.md`.. would be nice to not need hacks :)

cc @laurentlb re stardoc

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+    symbol_names = [
+        "rust_library",
+        "rust_binary",
+        "rust_benchmark",  # TODO(skydoc): no lint against absent symbols
+        "rust_test",
+        "rust_doc",
+        "rust_doc_test",
+        "rust_toolchain",
+        "rust_proto_toolchain",
+        "rust_proto_library",
```
Comment (damienmg) [inline docs/BUILD]: add rust_grpc_library?

File: proto/proto.bzl [hunk 2]
```diff
@@ -190,6 +190,10 @@ rust_proto_library = rule(
     _rust_proto_library_impl,
     attrs = {
         "deps": attr.label_list(
+            doc = """
+                List of proto_library dependencies that will be built.
+                One crate for each proto_library will be created with the corresponding stubs.
+            """,
```
Comment (damienmg) [inline proto/proto.bzl]: I know that this was not document so far, but maybe we should add comment to the rust_deps attribute too (it's the rust compile dependency for the rust_proto_library target, e.g. the proto library itself).

File: proto/proto.bzl [hunk 3]
```diff
@@ -241,12 +239,17 @@ rust_binary(
...
+)
 
 rust_grpc_library = rule(
     _rust_grpc_library_impl,
     attrs = {
         "deps": attr.label_list(
+            doc = """
+                List of proto_library dependencies that will be built.
+                One crate for each proto_library will be created with the corresponding gRPC stubs.
+            """,
```
Comment (damienmg) [inline proto/proto.bzl]: Same here for rust_deps

**Comment by laurentlb** [comment]
> Had various issues with stardoc

Please file bugs for this
Thanks!

**Pull request merged**